pkg/apis/operator/v1alpha1/validation/validation.go

```diff
@@ -128,6 +128,22 @@ func validateRuntimeCluster(runtimeCluster operatorv1alpha1.RuntimeCluster, fldP
 		}
 	}
 
+	if runtimeCluster.Ingress.Domain != "" {
+		allErrs = append(allErrs, gardencorevalidation.ValidateDNS1123Subdomain(runtimeCluster.Ingress.Domain, fldPath.Child("ingress", "domain"))...)
+	}
+
+	domains := sets.New[string]()
```

Comment:
In the old case domain was immutable, I guess it was done on purpose. Now, also first domain should be made immutable otherwise for example plutono domain will change again and again based on the first entry of the domains.

Reply:
Added more validation logic and corresponding tests in https://github.com/gardener/gardener/pull/9038/commits/5a5d31af42a290c656cfd11770ce6b8bd460add2.

I left this out initially as I do not see such a strong requirement to keep the domain static. Unlike the domain field of the virtual garden, which is tied to service account tokens, the ingress domains could actually change without large collateral damage. However, we can also add that later in case it becomes a requirement.